components/post_markdown/post_markdown.tsx

```diff
@@ -1,56 +1,56 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 
-import Markdown from 'components/markdown';
-
-import {renderSystemMessage} from './system_message_helpers.jsx';
-
-export default class PostMarkdown extends React.PureComponent {
-    static propTypes = {
-
-        /*
-         * Any extra props that should be passed into the image component
-         */
-        imageProps: PropTypes.object,
-
-        /*
-         * Whether or not this text is part of the RHS
-         */
-        isRHS: PropTypes.bool,
-
-        /*
-         * The post text to be rendered
-         */
-        message: PropTypes.string.isRequired,
-
-        /*
-         * The optional post for which this message is being rendered
-         */
-        post: PropTypes.object,
+import {Post} from 'mattermost-redux/types/posts';
+import {Channel} from 'mattermost-redux/types/channels';
 
-        /*
-         * The id of the channel that this post is being rendered in
-         */
-        channelId: PropTypes.string,
-
-        channel: PropTypes.object,
-
-        options: PropTypes.object,
-
-        pluginHooks: PropTypes.arrayOf(PropTypes.object),
-
-        /**
-         * Whether or not to place the LinkTooltip component inside links
-         */
-        hasPluginTooltips: PropTypes.bool,
+import Markdown from 'components/markdown';
 
-        isUserCanManageMembers: PropTypes.bool,
-        mentionKeys: PropTypes.array.isRequired,
-    };
+import {MentionKey, TextFormattingOptions} from 'utils/text_formatting';
+
+import {renderSystemMessage} from './system_message_helpers';
+
+type Props = {
+
+    /*
+     * Any extra props that should be passed into the image component
+     */
+    imageProps?: Record<string, any>;
+
+    /*
+     * Whether or not this text is part of the RHS
+     */
+    isRHS?: boolean;
+
+    /*
+     * The post text to be rendered
+     */
+    message: string;
+
+    /*
+     * The optional post for which this message is being rendered
+     */
+    post?: Partial<Post>;
+
+    /*
+     * The id of the channel that this post is being rendered in
+     */
+    channelId?: string;
+    channel?: Channel;
+    options?: TextFormattingOptions;
+    pluginHooks?: Array<Record<string, any>>;
+
+    /**
+     * Whether or not to place the LinkTooltip component inside links
+     */
+    hasPluginTooltips?: boolean;
+    isUserCanManageMembers?: boolean;
+    mentionKeys?: MentionKey[];
```

Comment:
This is required in the original component.

Reply:
this is related to the first comment.